### PR TITLE
fix(condo): DOMA-4112 changed submit and save logic for modal filters

### PR DESCRIPTION
**Pull request by Alllex202**
Changed submit and save logic for modal filters.
Added error and scrolling to field for 'template name'. 
Removed disabling 'save and submit' button. 

Before: if invalid template name then disabling "save and submit" button

After: if click "save and submit" button and template name is invalid then scrolling to field "template name" and showing error "it's field is required, bla-bla-bla..."

### CODE REVIEW

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx [hunk 1]
```diff
@@ -455,22 +470,28 @@ const Modal: React.FC<MultipleFiltersModalProps> = ({
...
         >
             <Input placeholder={NewTemplatePlaceholder} />
         </Form.Item>
-    ), [NewTemplateLabel, NewTemplatePlaceholder])
+    ), [FieldRequiredMessage, NewTemplateLabel, NewTemplatePlaceholder])
 
-    const handleSubmitButtonClick = useCallback(() => handleSaveRef.current(), [])
+    const handleSubmitButtonClick = useCallback(async () => {
+        const values = form.getFieldsValue()
+        if (values.hasOwnProperty(NON_FIELD_ERROR_NAME)) delete values[NON_FIELD_ERROR_NAME]
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:492]: maybe set null instead `delete`?

Comment (nomerdvadcatpyat) [reply apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:492]: ok

**Review by nomerdvadcatpyat — APPROVED** [review]

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx [hunk 2]
```diff
@@ -455,22 +470,28 @@ const Modal: React.FC<MultipleFiltersModalProps> = ({
             labelCol={LABEL_COL_PROPS}
             initialValue={get(selectedFiltersTemplate, 'name')}
             required
+            rules={[{ required: true, message: FieldRequiredMessage, whitespace: true }]}
```
Comment (akarjakin) [inline apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:473]: this array will be recreated on each render and have new address in memory

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx [hunk 3]
```diff
@@ -455,22 +470,28 @@ const Modal: React.FC<MultipleFiltersModalProps> = ({
...
         </Form.Item>
-    ), [NewTemplatePlaceholder, TemplateLabel, selectedFiltersTemplate])
+    ), [FieldRequiredMessage, NewTemplatePlaceholder, TemplateLabel, selectedFiltersTemplate])
 
     const NewFiltersTemplateNameInput = useCallback(() => (
         <Form.Item
             name='newTemplateName'
             label={NewTemplateLabel}
             labelCol={LABEL_COL_PROPS}
+            rules={[{ required: false, message: FieldRequiredMessage, whitespace: true }]}
```
Comment (akarjakin) [inline apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx:484]: this array will be recreated on each render and have new address in memory

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx (hunk 2 above)
Comment (Alllex202) [reply apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx]: done

File: apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx (hunk 3 above)
Comment (Alllex202) [reply apps/condo/domains/common/hooks/useMultipleFiltersModal.tsx]: done

**Review by akarjakin — APPROVED** [review]

**Pull request merged**